test/controllers/api/base_controller_test.rb

```diff
@@ -0,0 +1,85 @@
+require_relative './base_test_case'
+
+module API
+  class BaseControllerTest < API::BaseTestCase
+    test "log user for authenticated API request" do
+      travel_to(Time.utc(2022, 1, 5, 14, 53, 7))
+      redis = Exercism.redis_tooling_client
+      user_1 = create :user, handle: 'foo'
+      user_2 = create :user, handle: 'bar'
+
+      setup_user(user_1)
+      get api_docs_path, headers: @headers, as: :json
+      get api_docs_path, headers: @headers, as: :json
+      get api_docs_path, headers: @headers, as: :json
+      sign_out user_1
+
+      setup_user(user_2)
+      get api_docs_path, headers: @headers, as: :json
+
+      sleep(0.1)
```

Comment:
I tried stubbing `Thread.new`, but that lead to weird Redis errors caused by (I assume) them also using `Thread.new`.